### DISCUSSION TITLE
Difficulties with WASM bindings

### DISCUSSION THREAD

**Discussion by mr-martian** [tree-sitter/tree-sitter#2005] [Q&A] [answered]
I'm trying to make a webpage that does some analysis of some DSLs using tree-sitter.

```
├── index.html
└── wasm
    ├── tree-sitter-cg.wasm
    ├── tree-sitter.js
    ├── tree-sitter.wasm
    └── tree-sitter-xfst.wasm
```

I tried to follow the instructions in https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_web/README.md and ended up with `index.html` looking like this:
```html
<html>
  <head>
    <title>Apertium Rule-Explainer</title>
    <script src="wasm/tree-sitter.js"></script>
    <script>
      const TSParser = window.TreeSitter;
      TSParser.init().then(async function () {
          const CG = await TSParser.Language.load('wasm/tree-sitter-cg.wasm');
          console.log("HI");
      });
    </script>
  </head>
  <body>
  </body>
</html>
```
I've tried several variations of this, but the `TSParser.Language.load` part keeps resulting in something like
```
Uncaught (in promise) RuntimeError: abort(Assertion failed: undefined). Build with -s ASSERTIONS=1 for more info.
    at ee (tree-sitter.js:1:6630)
    at S (tree-sitter.js:1:4287)
    at Me (tree-sitter.js:1:9168)
    at tree-sitter.js:1:46338
```
I compiled the WASM bindings with `tree-sitter build-wasm` using
```
$ tree-sitter --version
tree-sitter 0.19.5 (8d8690538ef0029885c7ef1f163b0e32f256a5aa)
```
`npm upgrade` said there wasn't a newer version, so I also tried downgrading `tree-sitter.{js,wasm}` from 0.20.7 to 0.19.5, but to no avail.

Any ideas as to what I'm doing wrong here?

**Comment by sogaiu** [comment, marked as answer]
Possibly related? https://github.com/tree-sitter/tree-sitter/issues/1829

**Reply by mr-martian** [reply]
That was indeed the issue - the version requirements on emscripten should really be mentioned somewhere in the docs.
